### PR TITLE
merging branch homework3 to master

### PR DESCRIPTION
**Pull request by TaniaVoitsekhovska**

### CODE REVIEW

File: hw2.sql [hunk 1]
```diff
@@ -62,4 +63,38 @@ SELECT * FROM book ORDER BY number_of_pages DESC;
...
+SELECT * FROM book;
+UPDATE book SET author_surname='Тетчер' WHERE author_surname='Флетчер';
+UPDATE book SET author_surname='Блікс' WHERE id=23;
+SELECT COUNT(*) as number_of_books FROM book  WHERE author_awards IS NOT NULL;
+
+SELECT id, book_title, category, author_name, author_surname, number_of_pages
+FROM book
+WHERE number_of_pages = (SELECT MIN(number_of_pages) FROM book);
+
+SELECT id, MAX(number_of_pages) as Max_number_of_pages FROM book;
```
Comment (vtymtsiv) [inline hw2.sql]: This statement can't be executed, you can't show id with the result of `max()`.

> Error Code: 1140. In aggregated query without GROUP BY, expression 1 of SELECT list contains nonaggregated column 'library.book.id'; this is incompatible with sql_mode=only_full_group_by

File: hw2.sql [hunk 2]
```diff
@@ -62,4 +63,38 @@ SELECT * FROM book ORDER BY number_of_pages DESC;
...
+SELECT ROUND(AVG(book_price), 2) as average_price FROM book;
+SELECT id, category, SUM(book_price) as total_cost_of_books FROM book WHERE category = 'Художня проза';
```
Comment (vtymtsiv) [inline hw2.sql]: Same here

File: hw2.sql [hunk 3]
```diff
@@ -62,4 +63,38 @@ SELECT * FROM book ORDER BY number_of_pages DESC;
...
+SELECT id, MIN(book_price) as lowest_book_price FROM book;
```
Comment (vtymtsiv) [inline hw2.sql]: Same

**Review by vtymtsiv — CHANGES_REQUESTED** [review]
Some statements can't be executed.

File: hw2.sql [hunk 4]
```diff
@@ -62,4 +63,38 @@ SELECT * FROM book ORDER BY number_of_pages DESC;
...
+SELECT id, book_title, category FROM book WHERE category LIKE '%проза%';
+SELECT id, book_title, category, publication_date FROM book WHERE publication_date LIKE '2017%';
+SELECT id, book_title, book_price FROM book WHERE book_price BETWEEN 1.50 and 120.2;
+SELECT id, book_title, book_price FROM book WHERE book_price NOT BETWEEN 200 and 600;
+
+SET sql_safe_updates = 0;
+DELETE FROM book WHERE id = 2;
+DELETE FROM book WHERE author_name = 'Наталка';
+DELETE FROM book WHERE isbn = 9786176794394;
+SET sql_safe_updates = 1;
```
Comment (vtymtsiv) [inline hw2.sql]: It is not necessary to run all these statements with sql_safe_updates = 0

File: hw2.sql (hunk 1 above)
Comment (TaniaVoitsekhovska) [reply hw2.sql]: fixed

File: hw2.sql (hunk 2 above)
Comment (TaniaVoitsekhovska) [reply hw2.sql]: fixed

File: hw2.sql (hunk 3 above)
Comment (TaniaVoitsekhovska) [reply hw2.sql]: fixed

File: hw2.sql (hunk 4 above)
Comment (TaniaVoitsekhovska) [reply hw2.sql]: fixed

**Review by vtymtsiv — APPROVED** [review]

**Pull request merged**